client/trino-jdbc/src/main/java/io/trino/jdbc/PropertyName.java

```diff
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.jdbc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public enum PropertyName
+{
+    USER("user"),
+    PASSWORD("password"),
+    SESSION_USER("sessionUser"),
+    ROLES("roles"),
+    SOCKS_PROXY("socksProxy"),
+    HTTP_PROXY("httpProxy"),
+    APPLICATION_NAME_PREFIX("applicationNamePrefix"),
+    DISABLE_COMPRESSION("disableCompression"),
+    ASSUME_LITERAL_NAMES_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS("assumeLiteralNamesInMetadataCallsForNonConformingClients"),
+    ASSUME_LITERAL_UNDERSCORE_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS("assumeLiteralUnderscoreInMetadataCallsForNonConformingClients"),
+    SSL("SSL"),
+    SSL_VERIFICATION("SSLVerification"),
+    SSL_KEY_STORE_PATH("SSLKeyStorePath"),
+    SSL_KEY_STORE_PASSWORD("SSLKeyStorePassword"),
+    SSL_KEY_STORE_TYPE("SSLKeyStoreType"),
+    SSL_TRUST_STORE_PATH("SSLTrustStorePath"),
+    SSL_TRUST_STORE_PASSWORD("SSLTrustStorePassword"),
+    SSL_TRUST_STORE_TYPE("SSLTrustStoreType"),
+    SSL_USE_SYSTEM_TRUST_STORE("SSLUseSystemTrustStore"),
+    KERBEROS_SERVICE_PRINCIPAL_PATTERN("KerberosServicePrincipalPattern"),
+    KERBEROS_REMOTE_SERVICE_NAME("KerberosRemoteServiceName"),
+    KERBEROS_USE_CANONICAL_HOSTNAME("KerberosUseCanonicalHostname"),
+    KERBEROS_PRINCIPAL("KerberosPrincipal"),
+    KERBEROS_CONFIG_PATH("KerberosConfigPath"),
+    KERBEROS_KEYTAB_PATH("KerberosKeytabPath"),
+    KERBEROS_CREDENTIAL_CACHE_PATH("KerberosCredentialCachePath"),
+    KERBEROS_DELEGATION("KerberosDelegation"),
+    ACCESS_TOKEN("accessToken"),
+    EXTERNAL_AUTHENTICATION("externalAuthentication"),
+    EXTERNAL_AUTHENTICATION_TIMEOUT("externalAuthenticationTimeout"),
+    EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS("externalAuthenticationRedirectHandlers"),
+    EXTERNAL_AUTHENTICATION_TOKEN_CACHE("externalAuthenticationTokenCache"),
+    EXTRA_CREDENTIALS("extraCredentials"),
+    CLIENT_INFO("clientInfo"),
+    CLIENT_TAGS("clientTags"),
+    TRACE_TOKEN("traceToken"),
+    SESSION_PROPERTIES("sessionProperties"),
+    SOURCE("source"),
+    DNS_RESOLVER("dnsResolver"),
+    DNS_RESOLVER_CONTEXT("dnsResolverContext"),
+    // these two are not actual properties but parts of the path
+    CATALOG("catalog"),
+    SCHEMA("schema");
+
+    private final String key;
+
+    private static final Map<String, PropertyName> lookup = new HashMap<String, PropertyName>();
+
+    static {
+        for (PropertyName name : PropertyName.values()) {
+            lookup.put(name.toString(), name);
+        }
+    }
+
+    PropertyName(final String key)
+    {
+        this.key = key;
+    }
+
+    @Override
+    public String toString()
+    {
+        return key;
+    }
+
+    public static PropertyName get(String key)
```

Comment:
Instead, let's do make it:
```suggestion
    public static Optional<PropertyName> findByKey(String key)
```
so that method doesn't throw. Let's use if in `TrinoDriverUri` without try catch

Reply:
bump

Reply:
I haven't pushed the changes yet, since I want to address all comments first. I only replied to comments that didn't require changes.

Reply:
Done